src/trackinfoobject.cpp

```diff
@@ -43,103 +23,48 @@
-}
-
-TrackInfoObject::TrackInfoObject(const QDomNode &nodeHeader)
-        : m_qMutex(QMutex::Recursive),
-          m_analyserProgress(-1) {
-    QString filename = XmlParse::selectNodeQString(nodeHeader, "Filename");
-    QString location = QDir(XmlParse::selectNodeQString(nodeHeader, "Filepath")).filePath(filename);
-    m_fileInfo = QFileInfo(location);
-    m_pSecurityToken = Sandbox::openSecurityToken(m_fileInfo, true);
-
-    // We don't call initialize() here because it would end up calling parse()
-    // on the file. Plus those initializations weren't done before, so it might
-    // cause subtle bugs. This constructor is only used for legacy importing so
-    // I'm not going to do it. rryan 6/2010
-
-    m_sTitle = XmlParse::selectNodeQString(nodeHeader, "Title");
-    m_sArtist = XmlParse::selectNodeQString(nodeHeader, "Artist");
-    m_sType = XmlParse::selectNodeQString(nodeHeader, "Type");
-    m_sComment = XmlParse::selectNodeQString(nodeHeader, "Comment");
-    m_iDuration = XmlParse::selectNodeQString(nodeHeader, "Duration").toInt();
-    m_iSampleRate = XmlParse::selectNodeQString(nodeHeader, "SampleRate").toInt();
-    m_iChannels = XmlParse::selectNodeQString(nodeHeader, "Channels").toInt();
-    m_iBitrate = XmlParse::selectNodeQString(nodeHeader, "Bitrate").toInt();
-    m_iTimesPlayed = XmlParse::selectNodeQString(nodeHeader, "TimesPlayed").toInt();
-    m_fReplayGain = XmlParse::selectNodeQString(nodeHeader, "replaygain").toFloat();
-    m_bHeaderParsed = false;
-    m_bBpmLock = false;
-    m_Rating = 0;
-
-    // Mixxx <1.8 recorded track IDs in mixxxtrack.xml, but we are going to
-    // ignore those. Tracks will get a new ID from the database.
-    //m_id = XmlParse::selectNodeQString(nodeHeader, "Id").toInt();
-    m_id = TrackId();
-
-    m_fCuePoint = XmlParse::selectNodeQString(nodeHeader, "CuePoint").toFloat();
-    m_bPlayed = false;
-    m_bDeleteOnReferenceExpiration = false;
-    m_bDirty = false;
-    m_bLocationChanged = false;
-}
-
-void TrackInfoObject::initialize(bool parseHeader, bool parseCoverArt) {
-    m_bDeleteOnReferenceExpiration = false;
-    m_bDirty = false;
-    m_bLocationChanged = false;
-
-    m_sArtist = "";
-    m_sTitle = "";
-    m_sType= "";
-    m_sComment = "";
-    m_sYear = "";
-    m_sURL = "";
+          m_id(trackId) {
     m_iDuration = 0;
     m_iBitrate = 0;
     m_iTimesPlayed = 0;
     m_bPlayed = false;
-    m_fReplayGain = 0.;
+    m_fReplayGain = 0.0f;
     m_bHeaderParsed = false;
-    m_id = TrackId();
     m_iSampleRate = 0;
     m_iChannels = 0;
     m_fCuePoint = 0.0f;
     m_dateAdded = QDateTime::currentDateTime();
     m_Rating = 0;
     m_bBpmLock = false;
-    m_sGrouping = "";
-    m_sAlbumArtist = "";
+    m_analyserProgress = -1;
+}
 
-    // parse() parses the metadata from file. This is not a quick operation!
-    if (parseHeader) {
-        parse(parseCoverArt);
-    }
+//static
+TrackPointer TrackInfoObject::newTemporary(
```

Comment:
Because it's used for temporarily storing TIOs that are not managed by the 
cache. There are various places in Mixxx where this is needed. Now with the 
factory functions this becomes more obvious. Even newly added TIOs are first 
allocated as temporary objects before they are inserted into the database.

Only TIOs retrieved from the database will be managed by the cache. After 
all references to such a managed TIO have been dropped the modifications 
will be written both into the database and the tags in the audio file will 
be updated all at once to keep them synchronized.

On 08/31/2015 12:42 AM, Daniel Schürmann wrote:
>
> In src/trackinfoobject.cpp 
> <https://github.com/mixxxdj/mixxx/pull/695#discussion_r38276981>:
>
> >
> > -    // parse() parses the metadata from file. This is not a quick operation!
> > -    if (parseHeader) {
> > -        parse(parseCoverArt);
> > -    }
> > +//static
> > +TrackPointer TrackInfoObject::newTemporary(
>
> Wy we need this factory function nad why is it "Temporary"?
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/mixxxdj/mixxx/pull/695/files#r38276981>.
>